### PR TITLE
Test blocked resources warning

### PR DESCRIPTION
**Pull request by YairSlobodin1**

### CODE REVIEW

File: cmd/subcmds/main.go [hunk 1]
```diff
@@ -5,8 +5,23 @@ SPDX-License-Identifier: Apache-2.0
...
+
+func Main(args []string) (string, error) {
+	rootCmd := newRootCommand()
 	rootCmd.SetArgs(args[1:])
-	return rootCmd.Execute()
+	return cmdWrapper(rootCmd)
+}
+
+// also returns a warning as string
+func cmdWrapper(cmd *cobra.Command) (string, error) {
```
Comment (zivnevo) [inline cmd/subcmds/main.go:21]: Why two functions? Can't `cmdWrapper` be embedded in `Main`?

File: cmd/subcmds/synthACL.go [hunk 2]
```diff
@@ -11,15 +11,17 @@ import (
...
 	cmd := &cobra.Command{
 		Use:   "acl",
 		Short: "Generate Networks ACLs from connectivity specification",
 		Long: `Generate Network ACLs to only allow the specified connectivity, either for each subnet separately or per VPC.
 		Endpoints in the required-connectivity specification may be subnets, subnet segments, CIDR segments and externals.`,
 		Args: cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, _ []string) error {
-			return synthesis(cmd, args, synth.NewACLSynthesizer, args.singleacl, false)
+			warning, err := synthesis(cmd, args, synth.NewACLSynthesizer, args.singleacl, false)
+			cmd.Println(warning)
```
Comment (zivnevo) [inline cmd/subcmds/synthACL.go:23]: Move `Println` to `synthesis()`

File: test/tests_defs.go [hunk 3]
```diff
@@ -8,9 +8,10 @@ package test
 import "fmt"
 
 type testCase struct {
-	testName    string
-	args        *command
-	expectedErr string
+	testName       string
+	args           *command
+	expectedErr    string
+	blockedWarning *string
```
Comment (zivnevo) [inline test/tests_defs.go:14]:
```suggestion
	expectedWarning *string
```
Let's not assume all future warnings will be around blocked resources

File: test/synth_test_list.go [hunk 4]
```diff
@@ -218,24 +238,25 @@ func synthACLTestsList() []testCase {
...
 			testName: "acl_vpe_tf",
 			args: &command{
-				cmd:        synth,
+				cmd:        synthesis,
 				subcmd:     acl,
 				config:     sgTesting3Config,
 				spec:       aclVpeSpec,
 				outputFile: "%s/acl_vpe_tf/nacl_expected.tf",
 			},
+			blockedWarning: utils.Ptr("\n"),
```
Comment (zivnevo) [inline test/synth_test_list.go:247]: where does this new line come from?

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: test/synth_test_list.go (hunk 4 above)
Comment (YairSlobodin1) [reply test/synth_test_list.go]: `cmd.Println()`. changed to `cmd.Print()`.

File: pkg/synth/acl.go [hunk 5]
```diff
@@ -21,27 +21,28 @@ type ACLSynthesizer struct {
...
 }
 
 // makeACL translates Spec to a collection of nACLs
 // 1. generate nACL rules for relevant subnets for each connection
 // 2. generate nACL rules for blocked subnets (subnets that do not appear in Spec)
-func (a *ACLSynthesizer) makeACL() *ir.ACLCollection {
-	for _, conn := range a.spec.Connections {
+func (a *ACLSynthesizer) makeACL() (collection *ir.ACLCollection, warning string) {
+	for i := range a.spec.Connections {
+		conn := a.spec.Connections[i]
```
Comment (zivnevo) [inline pkg/synth/acl.go:40]: why this change?

**Review by zivnevo — CHANGES_REQUESTED** [review]

**Review by zivnevo — APPROVED** [review]
